monai/networks/nets/quicknat.py

```diff
@@ -0,0 +1,445 @@
+
+    def __init__(self, spatial_dims, in_channels, out_channels, strides, kernel_size, act=None, adn_ordering="A"):
+        super().__init__(spatial_dims, in_channels, out_channels, strides, kernel_size, adn_ordering, act)
+
+    def forward(self, input: torch.Tensor, weights=None, indices=None):
+        _, channel, *dims = input.size()
+        if weights is not None:
+            weights, _ = torch.max(weights, dim=0)
+            weights = weights.view(1, channel, 1, 1)
+            # use weights to adapt how the classes are weighted.
+            if len(dims) == 2:
+                out_conv = F.conv2d(input, weights)
+            else:
+                raise ValueError("Quicknat is a 2D architecture, please check your dimension.")
+        else:
+            out_conv = super().forward(input)
+        # no indices to return
+        return out_conv, None
+
+
+# Quicknat specific blocks. All blocks inherit from MONAI blocks but have adaptions to their structure
+class ConvConcatDenseBlock(ConvDenseBlock):
+    """
+    This dense block is defined as a sequence of 'Convolution' blocks. It overwrite the '_get_layer' methodto change the ordering of
+    Every convolutional layer is preceded by a batch-normalization layer and a Rectifier Linear Unit (ReLU) layer.
+    The first two convolutional layers are followed by a concatenation layer that concatenates
+    the input feature map with outputs of the current and previous convolutional blocks.
+    Kernel size of two convolutional layers kept small to limit number of paramters.
+    Appropriate padding is provided so that the size of feature maps before and after convolution remains constant.
+    The output channels for each convolution layer is set to 64, which acts as a bottle- neck for feature map selectivity.
+    The input channel size is variable, depending on the number of dense connections.
+    The third convolutional layer is also preceded by a batch normalization and ReLU,
+    but has a 1 * 1 kernel size to compress the feature map size to 64.
+    Args:
+        in_channles: variable depending on depth of the network
+        seLayer: Squeeze and Excite block to be included, defaults to None, valid options are {'NONE', 'CSE', 'SSE', 'CSSE'},
+        dropout_layer: Dropout block to be included, defaults to None.
+    :return: forward passed tensor
+    """
+
+    def __init__(
+        self,
+        in_channels: int,
+        se_layer: Optional[nn.Module] = None,
+        dropout_layer: Optional[type[nn.Dropout2d]] = None,
+        kernel_size: Sequence[int] | int = 5,
+        num_filters: int = 64,
+    ):
+        self.count = 0
+        super().__init__(
+            in_channels=in_channels,
+            spatial_dims=2,
+            # number of channels stay constant throughout the convolution layers
+            channels=[num_filters, num_filters, num_filters],
+            norm=("instance", {"num_features": in_channels}),
+            kernel_size=kernel_size,
+        )
```

Comment:
I included this workaround to avoid the error: "Incompatible default for argument". If there is a cleaner way to set nn.Identity in the constructor already I would be very happy to hear it :)

Reply:
I think you can use `type[nn.Module]` instead, because `Identity` is a class and `Identity()` is an instance:

```py
import torch

a: torch.nn.Module = torch.nn.Identity()
b: type[torch.nn.Module] = torch.nn.Identity

```